pkg/image/prune/prune.go

```diff
@@ -39,8 +39,12 @@ const (
 	// not keep an ImageNode from being a candidate for pruning.
 	WeakReferencedImageEdgeKind = "WeakReferencedImage"
 
+	// ReferencedImageConfigEdgeKind defines an edge from an ImageStreamNode or an
+	// ImageNode to an ImageComponentNode.
+	ReferencedImageConfigEdgeKind = "ReferencedImageConfig"
+
 	// ReferencedImageLayerEdgeKind defines an edge from an ImageStreamNode or an
-	// ImageNode to an ImageLayerNode.
+	// ImageNode to an ImageComponentNode.
 	ReferencedImageLayerEdgeKind = "ReferencedImageLayer"
```

Comment:
Well, I'm differentiating whether the image refers to the component as its *layer* or as its *config*. I'm using different edges for both cases. The edges don't serve any purpose later on. So instead of two, we could have just one `ReferencedImageComponent`, as you suggest. I kind of like the differentiation though.